src/runtime/minrpc/minrpc_logger.cc

```diff
@@ -0,0 +1,288 @@
+      break;
+    case kDLUInt:
+      LogVal<uint64_t>("(uint64)", value.v_int64);
+      break;
+    case kDLFloat: {
+      LogVal<float>("(float)", value.v_float64);
+      break;
+    }
+    case kTVMDataType: {
+      LogDLData("DLDataType(code,bits,lane)", &value.v_type);
+      break;
+    }
+    case kDLDevice: {
+      LogDLDevice("DLDevice(type,id)", &value.v_device);
+      break;
+    }
+    case kTVMPackedFuncHandle: {
+      LogVal<void*>("(PackedFuncHandle)", value.v_handle);
+      break;
+    }
+    case kTVMModuleHandle: {
+      LogVal<void*>("(ModuleHandle)", value.v_handle);
+      break;
+    }
+    case kTVMOpaqueHandle: {
+      LogVal<void*>("(OpaqueHandle)", value.v_handle);
+      break;
+    }
+    case kTVMDLTensorHandle: {
+      LogVal<void*>("(TensorHandle)", value.v_handle);
+      break;
+    }
+    case kTVMNDArrayHandle: {
+      LogVal<void*>("kTVMNDArrayHandle", value.v_handle);
+      break;
+    }
+    case kTVMNullptr:
+      LogString("Nullptr");
+      break;
+    case kTVMStr: {
+      LogString("\"");
+      LogString(value.v_str);
+      LogString("\"");
+      break;
+    }
+    case kTVMBytes: {
+      TVMByteArray* bytes = static_cast<TVMByteArray*>(value.v_handle);
+      int len = bytes->size;
+      LogVal<int64_t>("(Bytes) [size]: ", len);
+      if (PRINT_BYTES) {
+        LogString(", [Values]:");
+        LogString(" { ");
+        if (len > 0) {
+          LogVal<uint64_t>("", (uint8_t)bytes->data[0]);
+        }
+        for (int j = 1; j < len; j++) LogVal<uint64_t>(" - ", (uint8_t)bytes->data[j]);
+        LogString(" } ");
+      }
+      break;
+    }
+    default: {
+      LogString("ERROR-kUnknownTypeCode)");
+      break;
+    }
+  }
+  LogString("; ");
+}
+
+void Logger::OutputLog() {
+  LOG(INFO) << os_.str();
+  os_.str(std::string());
+}
+
+void MinRPCReturnsWithLog::ReturnVoid() {
+  next_->ReturnVoid();
+  logger_.LogString("-> ReturnVoid");
+  logger_.OutputLog();
+}
+
+void MinRPCReturnsWithLog::ReturnHandle(void* handle) {
+  next_->ReturnHandle(handle);
+  if (code_ == RPCCode::kGetGlobalFunc) {
+    RegisterHandleName(handle);
+  }
+  logger_.LogVal<void*>("-> ReturnHandle: ", handle);
+  logger_.OutputLog();
+}
+
+void MinRPCReturnsWithLog::ReturnException(const char* msg) {
+  next_->ReturnException(msg);
+  logger_.LogString("-> Exception: ");
+  logger_.LogString(msg);
+  logger_.OutputLog();
+}
+
+void MinRPCReturnsWithLog::ReturnPackedSeq(const TVMValue* arg_values, const int* type_codes,
+                                           int num_args) {
+  next_->ReturnPackedSeq(arg_values, type_codes, num_args);
+  ProcessValues(&arg_values, &type_codes, &num_args);
+  logger_.OutputLog();
+}
+
+void MinRPCReturnsWithLog::ReturnCopyAck(uint64_t* num_bytes, uint8_t** data_ptr) {
+  next_->ReturnCopyAck(num_bytes, data_ptr);
+  logger_.LogVal<uint64_t>("-> CopyAck: ", *num_bytes);
+  logger_.LogVal<void*>(", ", reinterpret_cast<void*>(*data_ptr));
```

Comment:
i think you can use static_cast<void*> here (and in general, try to use this instead of reinterpret_cast wherever possible)

Reply:
done